peps/pep-0747.rst

```diff
@@ -977,84 +604,67 @@ Consider the following possible pattern matching syntax:
-generally try to support *all* kinds of type expressions, so it doesn’t
-seem valuable to enumerate a particular subset.
+All functions observed in the wild that conceptually accept type form
+objects generally try to support *all* kinds of type expressions, so it
+doesn’t seem valuable to enumerate a particular subset.
 
-Additionally the above syntax isn’t precise enough to fully describe the
-actual input constraints for a typical function in the wild. For example
-many functions recognize un-stringified type expressions like
-``list[Movie]`` but may not recognize type expressions with stringified
-subcomponents like ``list['Movie']``.
+Additionally, the above syntax isn’t precise enough to fully describe the
+input constraints for a typical function in the wild. For example, many
+functions do not support type expressions with quoted subexpressions
+like ``list['Movie']``.
 
-A second use case for pattern matching on the interior of type
-expressions is to explicitly match an ``Annotated[]`` form to pull out the
-interior type argument and strip away the metadata:
+A second use case for pattern matching is to explicitly match an ``Annotated``
+form to extract the interior type argument and strip away any metadata:
 
 ::
 
    def checkcast(
-       typx: TypeExpr[T] | TypeExpr[AT=Annotated[T, *Anns]],
+       typx: TypeForm[T] | TypeForm[AT=Annotated[T, *A]],
        value: object
    ) -> T:
 
-However ``Annotated[T, metadata]`` is already treated equivalent to ``T`` anyway.
-There’s no additional value in being explicit about this behavior.
-The example above could be more-straightforwardly written as the equivalent:
+However, ``Annotated[T, metadata]`` is already treated equivalent to ``T``
+by static type checkers. There’s no additional value in being explicit about
+this behavior. The example above could more simply be written as the equivalent:
 
 ::
 
-   def checkcast(typx: TypeExpr[T], value: object) -> T:
-
-
-.. _recognize_uniontype_as_implicit_typeexpr_value:
-
-Recognize (T1 | T2) as an implicit TypeExpr value
--------------------------------------------------
-
-It would be nice if a value expression like ``int | str`` could be recognized
-as an implicit ``TypeExpr`` value and be used directly in a context where a
-``TypeExpr`` was expected. However making that possible would require making
-changes to the rules that type checkers use for the ``|`` operator. These rules
-are currently underspecified and would need to be make explicit first,
-before making changes to them. The PEP author is not sufficiently motivated to
-take on that specification work at the time of writing.
+   def checkcast(typx: TypeForm[T], value: object) -> T:
 
 
 Footnotes
 =========
 
-.. [#type_c]
-   :pep:`Type[C] spells a class object <484#the-type-of-class-objects>`
+.. [#type_t]
+   `Type[T] <https://typing.readthedocs.io/en/latest/spec/special-types.html#type>`__
+   spells a class object
```

Comment:
This one already had a reference, but it wasn't obvious what it was called. To check, go to https://typing.readthedocs.io/en/latest/spec/special-types.html#type and click "Show Source" and we can see:
```rst
.. _`type-brackets`:

``type[]``
----------

Sometimes you want to talk about class objects, in particular class
```

This shows `type-brackets` is the reference name: 


```suggestion
   :ref:`Type[T] <typing:type-brackets>` spells a class object
```